app/scripts/app.js

```diff
@@ -485,6 +485,81 @@ angular
       LabelFilter.readPersistedState();
     });
   })
+  .run(function($location, $uibModal, AuthService, Constants) {
+    if (!Constants.INACTIVITY_TIMEOUT_PERIOD > 0 || $location.path() === '/logout' || $location.path() === '/error') {
+      return false;
+    }
+    var isLocalLogoutModalShown = false;
+    var updateLastInteraction = function() {
+      localStorage.setItem('origin-web-console-last-interaction-timestamp', new Date());
```

Comment:
But the value of the latest interaction is set in different cases:
1, when the loading the page ,but it might happen that user wont interact with the page anymore so we need to set it by default
2, on user interaction
3, when dismissing the logout modal

Not really sure whats the benefit of setting a local var and updating the localStorage with it everytime, since the those cases might be mutual exclusive. Thought that having a simple helper that would just update the latestInteraction key in the local storage directly when any of those cases occur

Reply:
I'm talking about the key string, not the value.

Reply:
Treat it like a constant

Reply:
As in `var lastInteractionKey = "origin-web-console-last-interaction-timestamp";`

Reply:
You should be passing a string as the value to `setItem` not a `Date` object.